src/metrics/include/metrics/MetricsReporter.hpp

```diff
@@ -116,60 +81,158 @@ class ReporterT {
+                    }
+                }
             }
         }
     }
 
+    void writeGennySetupMetric(std::ostream& out, Permission perm) const {
+        const auto& ops = _registry->getOps(perm);
 
-    /**
-     * @return the number of data-points held by a map with values CounterImpl, GaugeImpl, etc.
-     */
-    template <class X>
-    static long dataPointsCount(const X& x, genny::metrics::v1::Permission perm) {
-        auto out = 0L;
-        for (const auto& v : x) {
-            out += v.second.getTimeSeries(perm).getDataPointCount(perm);
+        auto gennyOpsIt = ops.find("Genny");
+        if (gennyOpsIt == ops.end()) {
+            // We permit the Genny.Setup metric to be omitted to make unit testing easier.
+            return;
+        }
+
+        auto setupIt = gennyOpsIt->second.find("Setup");
+        if (setupIt == gennyOpsIt->second.end()) {
+            // We permit the Genny.Setup metric to be omitted to make unit testing easier.
+            return;
+        }
+
+        for (const auto& event : setupIt->second.at(0u)) {
+            out << nanosecondsCount(event.first.time_since_epoch());
+            out << ",";
+            out << "Genny.Setup";
+            out << ",";
+            out << nanosecondsCount(
+                static_cast<typename MetricsClockSource::duration>(event.second.duration));
+            out << std::endl;
+        }
+    }
+
+    void writeGennyActiveActorsMetric(std::ostream& out, Permission perm) const {
+        const auto& ops = _registry->getOps(perm);
+
+        auto gennyOpsIt = ops.find("Genny");
+        if (gennyOpsIt == ops.end()) {
+            // We permit the Genny.ActiveActors metric to be omitted to make unit testing easier.
+            return;
+        }
+
+        auto startedActorsIt = gennyOpsIt->second.find("ActorStarted");
+        if (startedActorsIt == gennyOpsIt->second.end()) {
+            // We permit the Genny.ActiveActors metric to be omitted to make unit testing easier.
+            return;
+        }
+
+        const auto& startedActors = startedActorsIt->second.at(0u);
+        const auto& finishedActors = gennyOpsIt->second.find("ActorFinished")->second.at(0u);
+
+        auto startedEventIt = startedActors.begin();
+        auto finishedEventIt = finishedActors.begin();
+
+        auto numActors = 0;
+        auto writeMetric = [&](const typename MetricsClockSource::time_point& when) {
+            out << nanosecondsCount(when.time_since_epoch());
+            out << ",";
+            out << "Genny.ActiveActors";
+            out << ",";
+            out << numActors;
+            out << std::endl;
+        };
+
+        // The termination condition of the while-loop is based only on `finishedActors` because
+        // there should always be as many ActorFinished events as there are ActorStarted events and
+        // an actor can only be finished after it has been started.
+        while (finishedEventIt != finishedActors.end()) {
```

Comment:
I admire your patience in writing this code considering it'll likely get tossed in your next PR 😸